leases/src/main/java/com/redhat/ipaas/leases/impl/SqlLeaseManager.java

```diff
@@ -0,0 +1,283 @@
+/**
+ * Copyright (C) 2016 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.redhat.ipaas.leases.impl;
+
+import com.redhat.ipaas.leases.Lease;
+import com.redhat.ipaas.leases.LeaseManager;
+import org.skife.jdbi.v2.DBI;
+import org.skife.jdbi.v2.Handle;
+import org.skife.jdbi.v2.PreparedBatch;
+import org.skife.jdbi.v2.util.LongColumnMapper;
+import org.skife.jdbi.v2.util.StringColumnMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+import java.sql.SQLException;
+import java.util.*;
+import java.util.concurrent.*;
+import java.util.function.Consumer;
+
+@Component
+public class SqlLeaseManager implements LeaseManager {
+
+    private final DBI dbi;
+    private final ConcurrentHashMap<String, SqlLease> leases = new ConcurrentHashMap<>();
+    protected ExecutorService executor = Executors.newSingleThreadExecutor();
+    protected ScheduledExecutorService scheduler = Executors.newSingleThreadScheduledExecutor();
+    private final String ownerId = UUID.randomUUID().toString();
+    private long leaseDuration = TimeUnit.SECONDS.toMillis(60);
+    private String insertSQL;
+
+    @Autowired
+    public SqlLeaseManager(DBI dbi) {
+        this.dbi = dbi;
+    }
+
+    @PostConstruct
+    public void start() {
+        executor = Executors.newSingleThreadExecutor();
+        scheduler = Executors.newSingleThreadScheduledExecutor();
+        withDB(dbi -> {
+            try {
+                long dbTime = 0;
+
+                // The insert SQL statement used will depend on the DB type that we are using.
```

Comment:
Two things here:

1. Can we extract the DB specific SQL and put it in the config? This so the code stays DB independent. 
2. I would use the DB as a dumb store and put the logic and time handling in Java. Time is hard.. we have to make sure this works across DST etc. I think there is a hard requirement that time is synced on all nodes on kubernetes.